mlflow/shap.py

```diff
@@ -32,3 +39,70 @@ def _log_matplotlib_figure(fig, out_file, artifact_path=None):
     """
     with _log_artifact_contextmanager(out_file, artifact_path) as tmp_path:
         fig.savefig(tmp_path)
+
+
+@experimental
+def log_explanation(predict_function, features, artifact_path=None):
+    """
+    Log the following items generated by SHAP.
+
+        - base values
+        - shap values
+        - summary bar plot
+
+    :param predict_function: A function to compute the output of a model.
+    :param features: A matrix of features to compute SHAP values with.
+    :param artifact_path: A run-relative artifact path the explanation is saved to. If unspecified,
+                          defaults to "shap".
+
+    :return: A URI of the logged explanation.
+
+    .. code-block:: python
+        :caption: Example
+
+        import pandas as pd
+        from sklearn.datasets import load_boston
+        from sklearn.linear_model import LinearRegression
+
+        import mlflow
+
+        # prepare training data
+        dataset = load_boston()
+        X = pd.DataFrame(dataset.data[:, :4], columns=dataset.feature_names[:4])
+        y = dataset.target
+
+        # train a model
+        model = LinearRegression()
+        model.fit(X, y)
+
+        # log an explanation
+        with mlflow.start_run() as run:
+            mlflow.shap.log_explanation(model.predict, X)
+
+        # print out logged items
+        client = mlflow.tracking.MlflowClient()
+        artifacts = [x.path for x in client.list_artifacts(run.info.run_id, "shap")]
+        print(artifacts)
+
+    .. code-block:: text
+        :caption: Output
+
+        ['shap/base_values.npy', 'shap/shap_values.npy', 'shap/summary_bar_plot.png']
```

Comment:
Nice! In the longer example/follow-up PR, we might want to show how to load back and use the logged base/shap values

Reply:
Should we provide a function to load back the logged values (e.g. `load_explanation`)?

Reply:
Could make sense yeah! I guess the alternative actually is not to use the explanation URI, but rather to remember the run ID & run-relative artifact path of the explanation, and call [download_artifacts](https://www.mlflow.org/docs/latest/python_api/mlflow.tracking.html#mlflow.tracking.MlflowClient.download_artifacts) with those parameters - it'd help to write out the workflow without `load_explanation`, and then we can consider adding an experimental `load_explanation` API

Reply:
I'll update the example using `download_artifacts` and later add `load_explanation` (if necessary) 👍